scipy/cluster/tests/test_hierarchy.py

```diff
@@ -125,17 +127,18 @@ def test_compare_with_trivial(self, xp):
         for method, code in _LINKAGE_METHODS.items():
             Z_trivial = _hierarchy.linkage(d, n, code)
             Z = linkage(xp.asarray(d), method)
-            assert_allclose(Z_trivial, Z, rtol=1e-14, atol=1e-15)
+            xp_assert_close(xp.asarray(Z_trivial), Z, rtol=1e-14, atol=1e-15)
```

Comment:
Which is calculated and which is expected? (Should these be reversed?)

Reply:
These are swapped now.